h/streamer/messages.py

```diff
@@ -91,33 +93,51 @@ def handle_annotation_event(message, socket):
     annotation event, otherwise a dict containing information about the event.
     """
     action = message['action']
-    annotation = storage.annotation_from_dict(message['annotation'])
 
     if action == 'read':
         return None
 
     if message['src_client_id'] == socket.client_id:
         return None
 
-    if annotation.get('nipsa') and (
-            socket.request.authenticated_userid != annotation.get('user', '')):
+    # We don't send anything until we have received a filter from the client
+    if socket.filter is None:
         return None
 
-    if not _authorized_to_read(socket.request, annotation):
+    notification = {
+        'type': 'annotation-notification',
+        'options': {'action': action},
+    }
+    id_ = message['annotation_id']
+
+    # Return early when action is delete
+    if action == 'delete':
+        notification['payload'] = [{'id': id_}]
+        return notification
```

Comment:
Hmm, just wanted to implement this and saw that it's going to be a bit more complicated for the `/stream` page. We do add the same filters as the page has (user, group, tag, etc.). It looks like the only viable solution for this is what we talked about not deleting annotations immediately but marking them as deleted and then later on actually removing them from the database. Which is quite a bit of work. Can you think of another solution?